resources/compute_instance_out.go

```diff
@@ -0,0 +1,349 @@
+package resources
+
+import (
+	"encoding/json"
+	"fmt"
+	conv "github.com/googleinterns/terraform-cost-estimation/memconverter"
+	"github.com/jedib0t/go-pretty/v6/table"
+	"github.com/jedib0t/go-pretty/v6/text"
+	"strings"
+)
+
+// The number of columns in output table for ComputeInstance.
+const numColumns = 5
+
+// ToTable creates a table.Table and fills it with the pricing information from
+// ComputeInstanceState.
+func (state *ComputeInstanceState) ToTable() (*table.Table, error) {
+	if state == nil {
+		return nil, nil
+	}
+	t := &table.Table{}
+	autoMerge := table.RowConfig{AutoMerge: true}
+
+	before := state.Before
+	after := state.After
+	if after == nil {
+		after = before
+	} else if before == nil {
+		before = after
+	}
+	if after == nil && before == nil {
+		return nil, fmt.Errorf("After and Before can't be nil at the same time.")
+	}
+	t.AppendRow(initRow("Name", before.Name, after.Name, false), autoMerge)
+	t.AppendRow(initRow("Instance_ID", before.ID, after.ID, true), autoMerge)
+	t.AppendRow(initRow("Zone", before.Zone, after.Zone, false), autoMerge)
+	t.AppendRow(initRow("Machine type", before.MachineType, after.MachineType, true), autoMerge)
+	//t.AppendRow(initRow("Core type", before.Core.Type, after.Core.Type, false), autoMerge)
+	//t.AppendRow(initRow("Memory type", before.Mem.Type, after.Mem.Type, true), autoMerge)
+
+	t.AppendRow(initRow("Action", state.Action, state.Action, false), autoMerge)
+	t.AppendRow(addHeader("Pricing\nInformation"), autoMerge)
+
+	core1, mem1, t1, err := getMemCoreInfo(state.Before)
```

Comment:
Maybe rename `t1` to `total1` because there's potential confusion about the different `t`s in this function.